commands/new_function.go

```diff
@@ -153,3 +168,73 @@ func printAvailableTemplates(availableTemplates []string) string {
 	}
 	return result
 }
+
+// insertAtRegexpMatch takes a full YAML file data (fileData), and
+// data to insert at the match location (insertData). Returns a string
+// that is fileData with functionYAMLData inserted in the 'functions:'
+// block of the YAML file.
+func insertFunctionIntoYAML(fileData string, functionYAMLData string) string {
+
+	// Create a regex with a capture group that grabs a string that starts with
+	// a non-whitespace character, ends with a colon, and is not a comment
+	// (e.g. 'provider:', 'functions:'). This should capture a top-level block
+	r := regexp.MustCompile(`(?m:^([^\#\s]+:)\s*$)`)
+	var retData string
+
+	// Get list of all matches, and list of all match indexes
+	strArr := r.FindAllStringSubmatch(fileData, -1)
+	intArr := r.FindAllStringSubmatchIndex(fileData, -1)
+
+	for i, matches := range strArr {
+
+		// If the capture group has matched with the 'functions:' block
+		if matches[1] == "functions:" {
+			if (len(strArr) - 1) > i {
+
+				// If the 'functions:' block is not the last top-level block,
+				// insert functionYAMLData right before the next block
+				insertLocation := intArr[i+1][0]
+				retData = fileData[:insertLocation] + functionYAMLData + fileData[insertLocation:]
+			} else {
+				// If the 'functions: ' block IS the last top-level block,
+				// append functionYAMLData to the end of the file
+				retData = fileData + functionYAMLData
+			}
+			break
+		}
+	}
+	return retData
+}
+
+func writeDockerfile(functionName string) error {
+	return ioutil.WriteFile("./"+functionName+"/Dockerfile", []byte(`FROM alpine:3.6
```

Comment:
Shouldn't be hardcoding this inline, at the very least pull it out into a variable - @alexellis does this belong outside of the cli?